Hi @heissanjay ,

- Can you fix the following errors?

```
AwsEc2ServiceImplTests.java:122: error: method launchAWSConfigurationTest in class AwsEc2ServiceImplTests cannot be applied to given types;
        launchAWSConfigurationTest(Settings.EMPTY, Protocol.HTTPS, null, -1, null, null, ClientConfiguration.DEFAULT_SOCKET_TIMEOUT);
        ^
  required: Settings,Protocol,String,int,Protocol,String,String,int
  found:    Settings,Protocol,<null>,int,<null>,<null>,int
  reason: actual and formal argument lists differ in length
/AwsEc2ServiceImplTests.java:137: error: incompatible types: String cannot be converted to Protocol
        launchAWSConfigurationTest(settings, Protocol.HTTP, "aws_proxy_host", 8080, "http",  "aws_proxy_username", "aws_proxy_password", 10000);
                                                                                    ^
```

- Can you run `./gradlew spotlessApply` (see [CONTRIBUTING](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#java-language-formatting-guidelines)) to fix some code style issues and push to the PR again?
- Also please run the `testThereIsADefaultClientByDefault` , `testAWSConfigurationWithAwsSettings`, and `Ec2DiscoveryPluginTests.java` tests (the ones this PR modifies) manually to ensure they work.

Afterwards I'll re-test your PR before approving.

--- a/plugins/discovery-ec2/src/test/java/org/elasticsearch/discovery/ec2/AwsEc2ServiceImplTests.java
+++ b/plugins/discovery-ec2/src/test/java/org/elasticsearch/discovery/ec2/AwsEc2ServiceImplTests.java
@@ -119,7 +119,16 @@ public class AwsEc2ServiceImplTests extends ESTestCase {
     }
 
     public void testAWSDefaultConfiguration() {
-        launchAWSConfigurationTest(Settings.EMPTY, Protocol.HTTPS, null, -1, null, null, ClientConfiguration.DEFAULT_SOCKET_TIMEOUT);
+        launchAWSConfigurationTest(
+            Settings.EMPTY,
+            Protocol.HTTPS,
+            null,
+            -1,
+            Protocol.HTTP,
+            null,
+            null,
+            ClientConfiguration.DEFAULT_SOCKET_TIMEOUT
+        );
     }
 
     public void testAWSConfigurationWithAwsSettings() {
@@ -134,7 +143,16 @@ public class AwsEc2ServiceImplTests extends ESTestCase {
             .put("discovery.ec2.read_timeout", "10s")
             .setSecureSettings(secureSettings)
             .build();
-        launchAWSConfigurationTest(settings, Protocol.HTTP, "aws_proxy_host", 8080, "http",  "aws_proxy_username", "aws_proxy_password", 10000);
+        launchAWSConfigurationTest(
+            settings,
+            Protocol.HTTP,
+            "aws_proxy_host",
+            8080,
+            Protocol.HTTP,
+            "aws_proxy_username",
+            "aws_proxy_password",
+            10000
+        );
     }
 
     protected void launchAWSConfigurationTest(

--- a/plugins/discovery-ec2/src/test/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryPluginTests.java
+++ b/plugins/discovery-ec2/src/test/java/org/elasticsearch/discovery/ec2/Ec2DiscoveryPluginTests.java
@@ -9,6 +9,7 @@
 package org.elasticsearch.discovery.ec2;
 
 import com.amazonaws.ClientConfiguration;
+import com.amazonaws.Protocol;
 import com.amazonaws.auth.AWSCredentials;
 import com.amazonaws.auth.AWSCredentialsProvider;
 import com.amazonaws.auth.BasicAWSCredentials;
@@ -196,7 +197,7 @@ public class Ec2DiscoveryPluginTests extends ESTestCase {
                     assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyPassword(), is("proxy_password_1"));
                     assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyHost(), is("proxy_host_1"));
                     assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyPort(), is(881));
-                    assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyProtocol(), is("http"));
+                    assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyProtocol(), is(Protocol.HTTP));
                     assertThat(((AmazonEC2Mock) clientReference.client()).endpoint, is("ec2_endpoint_1"));
                 }
                 // reload secure settings2
@@ -214,7 +215,7 @@ public class Ec2DiscoveryPluginTests extends ESTestCase {
                     assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyPassword(), is("proxy_password_1"));
                     assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyHost(), is("proxy_host_1"));
                     assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyPort(), is(881));
-                    assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyProtocol(), is("http"));
+                    assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyProtocol(), is(Protocol.HTTP));
                     assertThat(((AmazonEC2Mock) clientReference.client()).endpoint, is("ec2_endpoint_1"));
                 }
             }
@@ -232,7 +233,7 @@ public class Ec2DiscoveryPluginTests extends ESTestCase {
                 assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyPassword(), is("proxy_password_2"));
                 assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyHost(), is("proxy_host_2"));
                 assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyPort(), is(882));
-                assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyProtocol(), is("http"));
+                assertThat(((AmazonEC2Mock) clientReference.client()).configuration.getProxyProtocol(), is(Protocol.HTTP));
                 assertThat(((AmazonEC2Mock) clientReference.client()).endpoint, is("ec2_endpoint_2"));
             }
         }

